flink-streaming-connectors/flink-connector-elasticsearch2/src/main/java/org/apache/flink/streaming/connectors/elasticsearch2/ElasticsearchSink.java

```diff
@@ -177,7 +180,72 @@ public void open(Configuration configuration) {
 		if (LOG.isInfoEnabled()) {
 			LOG.info("Created Elasticsearch TransportClient {}", client);
 		}
+	}
+
+	@Override
+	public void invoke(final T element) {
+		ParameterTool params = ParameterTool.fromMap(userConfig);
```

Comment:
Its very inefficient to create a new `ParameterTool` instance for each incoming element.